### PR TITLE
Change return types in api

### PR DESCRIPTION
**Pull request by jczornik**
Return types are messed up in public API. This PR should fix it.

### CODE REVIEW

File: Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift [hunk 1]
```diff
@@ -6,9 +6,20 @@
...
 import Foundation
 
-/// Typealias for server response
+/// Typealias for raw response received from server
 typealias Response = (HTTPURLResponse, Data?)
 
+/// Typealias for body data received from server
+public typealias ServerData = Data?
+
+/// Parsed server response
```
Comment (travnick) [inline Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift:15]: add here more comments please

File: Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift [hunk 2]
```diff
@@ -21,7 +32,7 @@ protocol ResultDataSource: TriviallyConstructible {
     ///     - resultFor response: received response
     ///
     /// - Returns: Internal response object
```
Comment (jczornik) [inline Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift:34]: Return type in the comment does not correspond to the actual return type

File: Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift [hunk 3]
```diff
@@ -47,13 +58,13 @@ extension ResultDataSource {
...
```
Comment (jczornik) [inline Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift:60]: Return type in the comment does not correspond to the actual return type

File: Sources/JiraClient/ResponseHandlers/OkResponseHandler.swift [hunk 4]
```diff
@@ -18,19 +18,14 @@ struct OkResponseHandler<T>: ResponseHandler where T: DecodableDTO {
     ///     - RequestResult if succeeded.
     ///     - If response code is not 200 will return nil.
     ///     - If cannot parse response body will return .failure(.CannotParseResponse)
```
Comment (jczornik) [inline Sources/JiraClient/ResponseHandlers/OkResponseHandler.swift:20]: Return type in the comment does not correspond to the actual return type

File: Sources/JiraClient/ResponseHandlers/UnauthorisedResponseHandler.swift [hunk 5]
```diff
@@ -19,18 +19,14 @@ struct UnauthorizedResponseHandler: ResponseHandler {
     ///     - .failure(.AccessDenied) if succeeded.
     ///     - If response code is not 401 will return nil.
     ///     - If cannot parse response body will return .failure(.CannotParseResponse)
```
Comment (jczornik) [inline Sources/JiraClient/ResponseHandlers/UnauthorisedResponseHandler.swift:21]: Return type in the comment does not correspond to the actual return type

File: Tests/JiraClientTests/Mocks/ServerRequesting/ServerRequestingDummy.swift [hunk 6]
```diff
@@ -17,8 +17,8 @@ struct ServerRequestingDummy: ServerRequesting {
 
     var headers = [RequestHeader]()
 
-    func send(request: URLRequest) -> Future<RequestResult, Error> {
-        Future<RequestResult, Error> { _ in
+    func send(request: URLRequest) -> Future<ServerResponse, ServerCommunicationError> {
+        Future<ServerResponse, ServerCommunicationError> { _ in
 
         }
```
Comment (jczornik) [inline Tests/JiraClientTests/Mocks/ServerRequesting/ServerRequestingDummy.swift:23]: This should not be empty

File: Tests/JiraClientTests/ResponseHandlers/UnauthorizedResponseHandlerTests.swift [hunk 7]
```diff
@@ -16,39 +16,28 @@ class UnauthorizedResponseHandlerTests: XCTestCase {
...
+        let handler = UnauthorisedResponseHandler()
         let result = handler.handle(response: (unauthorizedResponse, nil))
 
-        if case .failure(let error) = result {
-            XCTAssert(error == .AccessDenied)
+        if case .Unauthorised(_) = result {
+            return
         } else {
             XCTFail("Result should be failure")
         }
```
Comment (jczornik) [inline Tests/JiraClientTests/ResponseHandlers/UnauthorizedResponseHandlerTests.swift:26]: This can be simplified

**Review by jczornik — COMMENTED** [review]
Issues to fix before merge

**Review by travnick — CHANGES_REQUESTED** [review]
review done

File: Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift (hunk 2 above)
Comment (jczornik) [reply Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift]: Fixed

File: Sources/JiraClient/ResponseHandlers/OkResponseHandler.swift (hunk 4 above)
Comment (jczornik) [reply Sources/JiraClient/ResponseHandlers/OkResponseHandler.swift]: Fixed

File: Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift (hunk 3 above)
Comment (jczornik) [reply Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift]: Fixed

File: Sources/JiraClient/ResponseHandlers/UnauthorisedResponseHandler.swift (hunk 5 above)
Comment (jczornik) [reply Sources/JiraClient/ResponseHandlers/UnauthorisedResponseHandler.swift]: Fixed

File: Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift (hunk 1 above)
Comment (jczornik) [reply Sources/JiraClient/Clients/BaseClient/ResultDataSource.swift]: Fixed

**Pull request merged**